### PR TITLE
Cover/cmds/core/ping

### PR DESCRIPTION
**Pull request by llogen**
Refactored ping.go and added tests for it.

Signed-off-by: llogen <christophlange95@googlemail.com>

### CODE REVIEW

**Comment by llogen** [comment]
Ein Test failed noch auf Grund einer Hardgecodeten IP, da kümmere ich mich morgen früh drum.

File: cmds/core/ping/ping.go [hunk 1]
```diff
@@ -160,32 +160,43 @@ func ping1(net6 bool, host string, i uint64, waitFor time.Duration) (string, err
...
+func main() {
+	flag.Parse()
+	if *version {
+		fmt.Fprintf(os.Stdout, "ping utility, Uroot version\n")
+		os.Exit(0)
+	}
+	// options without parameters (right now just: -hV)
+	if flag.NArg() < 1 {
+		flag.Usage()
+		os.Exit(0)
```
Comment (MDr164) [inline cmds/core/ping/ping.go:196]: I guess this case can be os.Exit(1) as the user did not provide enough arguments, causing the program to bail out and display proper usage.

File: cmds/core/ping/ping.go [hunk 2]
```diff
@@ -160,32 +160,43 @@ func ping1(net6 bool, host string, i uint64, waitFor time.Duration) (string, err
...
+	}
+	host := flag.Args()[0]
+	if err := ping(host); err != nil {
+		log.Fatalf("%v", err)
```
Comment (MDr164) [inline cmds/core/ping/ping.go:200]: Should be log.Fatal(err)

File: cmds/core/ping/ping_test.go [hunk 3]
```diff
@@ -0,0 +1,234 @@
+// Copyright 2015-2017 the u-root Authors. All rights reserved
```
Comment (MDr164) [inline cmds/core/ping/ping_test.go:1]: 2021

File: cmds/core/ping/ping_test.go [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+			},
+			net6:    false,
+			host:    "test.com",
+			i:       1,
+			waitFor: time.Minute,
+			exp:     fmt.Errorf("wrong sequence number %v (expected %v)", 0, 1),
+		},
+	} {
+		t.Run(tt.name, func(t *testing.T) {
+
```
Comment (MDr164) [inline cmds/core/ping/ping_test.go:180]: blank line

**Review by MDr164 — CHANGES_REQUESTED** [review]

File: cmds/core/ping/ping_test.go [hunk 5]
```diff
@@ -0,0 +1,233 @@
...
+			audible:    true,
+			waitFor:    time.Minute,
+			exp:        fmt.Errorf("ping failed: %v", fmt.Errorf("net.Dial(%v %v) failed: %v", "ip4:icmp", "", "dial ip4:icmp: missing address")),
+		},
+	} {
+		t.Run(tt.name, func(t *testing.T) {
+			*packetSize = tt.packetSize
+			*audible = tt.audible
+			err := ping(tt.host)
+			if err != nil {
```
Comment (ChriMarMe) [inline cmds/core/ping/ping_test.go:221]: in eine zeile :)

File: cmds/core/ping/ping_test.go [hunk 6]
```diff
@@ -0,0 +1,232 @@
...
+			waitFor:    time.Minute,
+			want:       fmt.Errorf("packet size too small (must be >= 8): %v", 7),
+		},
+		{
+			name:       "ping with error",
+			packetSize: 8,
+			host:       "",
+			audible:    true,
+			waitFor:    time.Minute,
+			want:       fmt.Errorf("ping failed: %v", fmt.Errorf("net.Dial(%v %v) failed: %v", "ip4:icmp", "", "dial ip4:icmp: missing address")),
```
Comment (walterchris) [inline cmds/core/ping/ping_test.go:214]: Why all these Errorf? It's static here and will not change will it?

Comment (llogen) [reply cmds/core/ping/ping_test.go:214]: No it won't

**Pull request closed without merging**